core/src/main/java/apoc/trigger/TriggerHandlerWrite.java

```diff
@@ -0,0 +1,153 @@
+package apoc.trigger;
+
+import apoc.SystemLabels;
+import apoc.SystemPropertyKeys;
+import apoc.util.Util;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.internal.helpers.collection.Pair;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static apoc.ApocConfig.APOC_TRIGGER_ENABLED;
+import static apoc.ApocConfig.apocConfig;
+
+public class TriggerHandlerWrite {
+    public static final String NOT_ENABLED_ERROR = "Triggers have not been enabled." +
+            " Set 'apoc.trigger.enabled=true' in your apoc.conf file located in the $NEO4J_HOME/conf/ directory.";
+
+    public static Map<String, Object> toTriggerInfo(Node node) {
+        return node.getAllProperties()
+                .entrySet().stream()
+                .filter(e -> !List.of(SystemPropertyKeys.name.name(), SystemPropertyKeys.database.name()).contains(e.getKey()))
+                .collect(HashMap::new, // workaround for https://bugs.openjdk.java.net/browse/JDK-8148463
+                        (mapAccumulator, e) -> {
+                            Object value = List.of(SystemPropertyKeys.selector.name(), SystemPropertyKeys.params.name()).contains(e.getKey()) 
+                                    ? Util.fromJson((String) e.getValue(), Map.class) 
+                                    : e.getValue();
+                            
+                            mapAccumulator.put(e.getKey(), value);
+                    }, HashMap::putAll);
+    }
+
+    private static boolean isEnabled() {
+        return apocConfig().getBoolean(APOC_TRIGGER_ENABLED);
+    }
+
+    public static void checkEnabled() {
+        if (!isEnabled()) {
+            throw new RuntimeException(NOT_ENABLED_ERROR);
+        }
+    }
+
+    public static Map<String, Object> install(String databaseName, String triggerName, String statement, Map<String,Object> selector, Map<String,Object> params) {
+        checkEnabled();
+        final HashMap<String, Object> previous = new HashMap<>();
+
+        withSystemDb(tx -> {
```

Comment:
There are many times where we should already be in a transaction being executed against "system" where we are opening a new nested transaction.

If it's not necessary then I don't think we should be opening nested transactions because these are known to have some issues. It's also quite expensive to open a new transaction, compared to reusing an existing one.

If this code is necessary then that changes things, but I'd like to know why that is the case!

Reply:
This is still a pending point @vga91. I think we check we are executing this against the system database in the functionality that uses it? So the system transaction is not needed

Reply:
Just to be sure, do you mean using a `@Context Transaction tx` instead?
Because in this case I receive an error `"Creating new node label on database 'system' is not allowed for user '' with FULL overridden by READ overridden by READ. See GRANT CREATE NEW NODE LABEL ON DATABASE system "`

Reply:
Let's do this as a separate piece of work, there's a ticket in Trello to reduce the number of transactions we use